File: mobile/src/main/java/treehou/se/habit/sitemaps/SitemapListComponent.java
package treehou.se.habit.sitemaps;


import dagger.Subcomponent;
import treehou.se.habit.main.MainActivityModule;
import treehou.se.habit.module.ActivityScope;
import treehou.se.habit.module.FragmentComponent;
import treehou.se.habit.module.FragmentComponentBuilder;
import treehou.se.habit.ui.sitemaps.SitemapListFragment;

@ActivityScope
@Subcomponent(
        modules = SitemapListModule.class
)
public interface SitemapListComponent extends FragmentComponent<SitemapListFragment> {

    @Subcomponent.Builder
    interface Builder extends FragmentComponentBuilder<SitemapListModule, SitemapListComponent> {

    }
}

File: mobile/src/main/java/treehou/se/habit/dagger/fragment/SitemapListComponent.kt
package treehou.se.habit.dagger.fragment

import dagger.Subcomponent
import treehou.se.habit.dagger.FragmentComponent
import treehou.se.habit.dagger.FragmentComponentBuilder
import treehou.se.habit.dagger.scopes.ActivityScope
import treehou.se.habit.ui.sitemaps.sitemaplist.SitemapListFragment

@ActivityScope
@Subcomponent(modules = arrayOf(SitemapListModule::class))
interface SitemapListComponent : FragmentComponent<SitemapListFragment> {

    @Subcomponent.Builder
    interface Builder : FragmentComponentBuilder<SitemapListModule, SitemapListComponent>
}